lib/vanagon/component.rb

```diff
@@ -290,6 +290,10 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      { name => { 'version' => version, 'ref' => options[:ref] }.delete_if { |k, v| !v } }
```

Comment:
I wouldn't even purge empty values since JSON serialization understands `nil` as `null`. That'd just look like this:

```ruby
>> puts JSON.pretty_generate({ 'rando' => { 'version' => '1.2.3', 'ref' => nil }})
{
  "rando": {
    "version": "1.2.3",
    "ref": null
  }
}
>> JSON.parse(JSON.pretty_generate({ 'rando' => { 'version' => '1.2.3', 'ref' => nil }}))
=> {
  "rando" => {
    "version" => "1.2.3",
        "ref" => nil
  }
}
```
----

Also, I strongly disagree that the suggested oneliner is unreadable. That's a very, very common Ruby construction. It could be spread over a few lines if it really bothers you:

```ruby
{ name => { 'version' => version, 'ref' => options[:ref] }
  .delete_if { |k, v| !v } }

# or

{ 
  name => {
    'version' => version, 
    'ref' => options[:ref]
  }.delete_if { |k, v| !v }
}